### PR TITLE
Created controllers and pages for work with manufacturer, car and driver

### PR DESCRIPTION
**Pull request by YuraMitriakhin**

### CODE REVIEW

File: src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            Driver driver = driverService.get(driverId);
+            car.getDrivers().add(driver);
+            carService.update(car);
+            resp.sendRedirect(req.getContextPath() + "/");
+        } catch (NoSuchElementException e) {
+            req.setAttribute("message", "Wrong car id " + carId + " or driver id "
+                    + driverId + " please input correct id!");
+            req.getRequestDispatcher("/WEB-INF/views/addDriverToCar.jsp")
+                    .forward(req, resp);
+        }
```
Comment (Max-Pochepets) [inline src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: this is redundant in the scope of this task. assume that the date will always be relevant

File: src/main/java/com/gmail/yuramitryahin/controllers/RegistrationCarController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.gmail.yuramitryahin.service.CarService;
+import com.gmail.yuramitryahin.service.ManufacturerService;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationCarController extends HttpServlet {
```
Comment (Max-Pochepets) [inline src/main/java/com/gmail/yuramitryahin/controllers/RegistrationCarController.java]:
```suggestion
public class CreateCarController extends HttpServlet {
```

File: src/main/webapp/WEB-INF/views/addDriver.jsp [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Driver</title>
+</head>
+<body>
+<form method="post" action="${pageContext.request.contextPath}/driver/register">
+    Input name <input type="text" name="name" placeholder="name" required>
```
Comment (Max-Pochepets) [inline src/main/webapp/WEB-INF/views/addDriver.jsp]:
```suggestion
    Input name <input type="text" name="driver_name" placeholder="name" required>
```

File: src/main/webapp/WEB-INF/views/addDriverToCar.jsp [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
+</head>
+<body>
+<h4 style="color:red">${message}</h4>
+<form method="post" action="${pageContext.request.contextPath}/driver/register/to/car">
+    Input car id <input type="number" name="carId" placeholder="car id" required>
+    <br>Input driver id <input type="number" name="driverId" placeholder="driver id" required>
```
Comment (Max-Pochepets) [inline src/main/webapp/WEB-INF/views/addDriverToCar.jsp]:
```suggestion
    <br>Input driver id <input type="number" name="driver_id" placeholder="driver id" required>
```

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 5]
```diff
@@ -5,5 +5,11 @@
 </head>
 <body>
 <h1>Hello world!</h1>
+<br><h3><a href="${pageContext.request.contextPath}/drivers/all">All drivers</a>
+<a href="${pageContext.request.contextPath}/driver/add">Add driver</a>
+<a href="${pageContext.request.contextPath}/manufacturer/add">Add manufacturer</a>
+    <a href="${pageContext.request.contextPath}/car/add">Add car</a>
+    <a href="${pageContext.request.contextPath}/driver/add/to/car">Add car</a>
```
Comment (Max-Pochepets) [inline src/main/webapp/WEB-INF/views/index.jsp]:
```suggestion
    <a href="${pageContext.request.contextPath}/drivers/cars/add">Add car</a>
```

File: src/main/webapp/web.xml [hunk 6]
```diff
@@ -7,8 +7,79 @@
...
\ No newline at end of file
+
+    <servlet>
+        <servlet-name>drivers</servlet-name>
+        <servlet-class>com.gmail.yuramitryahin.controllers.DriversController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>drivers</servlet-name>
+        <url-pattern>/drivers/all</url-pattern>
```
Comment (Max-Pochepets) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/</url-pattern>
```
just a suggestion

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]
make sure you have an empty line at the end of all files

**Comment by Max-Pochepets** [comment]
and keep your `package` names single

File: src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java (hunk 1 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: I think redundant  - it isn't bad

Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: in the task does not write that the data will always be correct

File: src/main/webapp/WEB-INF/views/addDriver.jsp (hunk 3 above)
Comment (YuraMitriakhin) [reply src/main/webapp/WEB-INF/views/addDriver.jsp]: fixed

File: src/main/java/com/gmail/yuramitryahin/controllers/RegistrationCarController.java (hunk 2 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controllers/RegistrationCarController.java]: fixed

File: src/main/webapp/WEB-INF/views/addDriverToCar.jsp (hunk 4 above)
Comment (YuraMitriakhin) [reply src/main/webapp/WEB-INF/views/addDriverToCar.jsp]: fixed

File: src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java (hunk 1 above)
Comment (Max-Pochepets) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: > in the task does not write that the data will always be correct

your current task is to implement your services on  web-server. in a real project frontend team is responsible for input validation. don't overcomplicate.

File: src/main/webapp/WEB-INF/views/index.jsp (hunk 5 above)
Comment (YuraMitriakhin) [reply src/main/webapp/WEB-INF/views/index.jsp]: fixed

File: src/main/webapp/web.xml (hunk 6 above)
Comment (YuraMitriakhin) [reply src/main/webapp/web.xml]: fixed

File: src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java (hunk 1 above)
Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: frontend team doesn't know about data in database)

Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: in real project we should sent request to frontend (maybe code that which will mean that the object was not found)

**Review by Max-Pochepets — APPROVED** [review]
there is no such point as `Validate input for presence in Data Base` in "documentation". there is only `check if any field is empty or that field should accept only numbers`. But ok.

File: src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: normally user doesn't not know about any id, he will just choose some item from the list, and its id will be sent to backend. And this list has been rendered to users based on the data that the frontend got from the backend, so the id should be proper. Also catching runtime exceptions means some problem with the code.

Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: okey, I understood and fised

Comment (YuraMitriakhin) [reply src/main/java/com/gmail/yuramitryahin/controllers/AddDriverToCarController.java]: fixed*

File: src/main/java/com/gmail/yuramitryahin/controller/RegistrationManufacturerController.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import com.gmail.yuramitryahin.lib.Injector;
+import com.gmail.yuramitryahin.modal.Manufacturer;
+import com.gmail.yuramitryahin.service.ManufacturerService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationManufacturerController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/gmail/yuramitryahin/controller/RegistrationManufacturerController.java:12]:
```suggestion
public class CreateManufacturerController extends HttpServlet {
```
controller name should starts with the verb, let's change it in each of them

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -5,10 +5,81 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>registrationDrivers</servlet-name>
+        <servlet-class>com.gmail.yuramitryahin.controller.RegistrationDriverController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>registrationDrivers</servlet-name>
+        <url-pattern>/driver/add</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:33]:
```suggestion
        <url-pattern>/drivers/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 9]
```diff
@@ -5,10 +5,81 @@
...
+    </servlet-mapping>
+
+    <servlet-mapping>
+        <servlet-name>registrationDrivers</servlet-name>
+        <url-pattern>/driver/register</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:38]:
```suggestion
        <url-pattern>/drivers/register</url-pattern>
```

File: src/main/webapp/web.xml [hunk 10]
```diff
@@ -5,10 +5,81 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>registrationManufacturer</servlet-name>
+        <servlet-class>com.gmail.yuramitryahin.controller.RegistrationManufacturerController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>registrationManufacturer</servlet-name>
+        <url-pattern>/manufacturer/add</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:48]:
```suggestion
        <url-pattern>/manufacturers/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -5,10 +5,81 @@
...
+    </servlet-mapping>
+
+    <servlet-mapping>
+        <servlet-name>registrationManufacturer</servlet-name>
+        <url-pattern>/manufacturer/register</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:53]:
```suggestion
        <url-pattern>/manufacturers/register</url-pattern>
```

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -5,10 +5,81 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>registrationCar</servlet-name>
+        <servlet-class>com.gmail.yuramitryahin.controller.CreateCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>registrationCar</servlet-name>
+        <url-pattern>/car/add</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:63]:
```suggestion
        <url-pattern>/cars/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -5,10 +5,81 @@
...
+    </servlet-mapping>
+
+    <servlet-mapping>
+        <servlet-name>registrationCar</servlet-name>
+        <url-pattern>/car/register</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:68]:
```suggestion
        <url-pattern>/cars/register</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -5,10 +5,81 @@
...
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/drivers/cars/add</url-pattern>
+    </servlet-mapping>
+
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/driver/car/register</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:83]:
```suggestion
        <url-pattern>/cars/drivers/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -5,10 +5,81 @@
...
+
+    <servlet>
+        <servlet-name>addDriverToCar</servlet-name>
+        <servlet-class>com.gmail.yuramitryahin.controller.AddDriverToCarController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addDriverToCar</servlet-name>
+        <url-pattern>/drivers/cars/add</url-pattern>
+    </servlet-mapping>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:79]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**